openmetadata-service/src/main/java/org/openmetadata/service/security/auth/BasicAuthenticator.java

```diff
@@ -523,7 +523,7 @@ public void validatePassword(String providedIdentity, User storedUser, String re
 
   @Override
   public User lookUserInProvider(String userName) {
-    User storedUser;
+    User storedUser = null;
     try {
       if (userName.contains("@")) {
```

Comment:
do we need check this condition anymore? we should enforce that at JwtFilter and throw an exception if there is no email is present

Reply:
problem is in existing flow we used to have principal claims, where even if someone used [sub] claim first then we built the email ourself as [sub@principalDomain] , if I do as suggested above then it will break the existing user flow since he initially used [sub or maybe preferred_username] that itself is not an email (email is what we built) , on new claim mapping have added that check (see below). but for not breaking the existing flow I have not done that for earlier flow

<img width="935" alt="image" src="https://github.com/open-metadata/OpenMetadata/assets/105265192/7cf99c60-f4c6-428f-b230-623582bf70fd">